### PR TITLE
Add settings system on frontend

### LINKED ISSUE #630: core: frontend: Make settings persistent
We should create a settings system that can save, load and restore settings 

E.g: Dark theme will not work when the window opens again or the system restart 
https://github.com/bluerobotics/companion-docker/pull/629

### PR DESCRIPTION
**Pull request by patrickelectric**
- Fix #630

### CODE REVIEW

**Comment by Williangalvani** [comment]
lol. what's up with CI this time? :man_facepalming:

**Comment by patrickelectric** [comment]
Probably related:
- https://github.com/samuelcolvin/pydantic/issues/3528

File: core/frontend/src/store/settings.ts [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+  getModule, Module, Mutation, VuexModule,
+} from 'vuex-module-decorators'
+
+import store from '@/store'
+
+  @Module({
+    dynamic: true,
+    store,
+    name: 'settings',
+  })
```
Comment (rafaellehmkuhl) [inline core/frontend/src/store/settings.ts:12]: Strange indent. How that passed?

**Review by rafaellehmkuhl — CHANGES_REQUESTED** [review]
Cool! 🐠

File: core/frontend/src/store/settings.ts (hunk 1 above)
Comment (patrickelectric) [reply core/frontend/src/store/settings.ts]: Done.

File: core/frontend/src/libs/settings.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  // eslint-disable-next-line
+  get is_dark_theme(): boolean {
+    vuetify.framework.theme.dark = settingsStore.is_dark_theme
+    return settingsStore.is_dark_theme
+  }
+
+  // eslint-disable-next-line
+  set is_dark_theme(value: boolean) {
+    settingsStore.setDarkTheme(value)
+  }
```
[the remark this answers is not in the document]
Comment (patrickelectric) [reply core/frontend/src/libs/settings.ts:14]: When storage is updated (or set is used), the get will be called automatically.

**Review by rafaellehmkuhl — APPROVED** [review]

**Pull request merged**